benches/search_comparison.rs

```diff
@@ -285,3 +257,17 @@ fn make_sorted_vec<T: Ord>(mut v: Vec<T>) -> Vec<T> {
 fn search_sorted_vec<'a, T: Ord>(c: &'a Vec<T>, x: T) -> Option<&'a T> {
     c.binary_search(&x).ok().map(|i| &c[i])
 }
+
+fn pseudorandom_iter<T>(mut seed: usize, max: usize) -> impl Iterator<Item = T>
+where
+    T: TryFrom<usize>,
+    <T as TryFrom<usize>>::Error: core::fmt::Debug,
+{
+    std::iter::from_fn(move || {
+        // LCG constants from https://en.wikipedia.org/wiki/Numerical_Recipes.
+        seed = seed.wrapping_mul(1664525).wrapping_add(1013904223);
+        let r = seed % max;
+
+        Some(black_box(T::try_from(r).unwrap()))
```

Comment:
I don't think we need `black_box()` here, because it's quite general code. It's seems like it was just moved when refactoring the code.

Reply:
done